ci-operator/step-registry/enable-qe-catalogsource/logging/enable-qe-catalogsource-logging-commands.sh

```diff
@@ -1,226 +1,601 @@
+    source: registry.redhat.io/openshift-logging/kibana6-rhel8
+  - mirrors:
+    - quay.io/redhat-user-workloads/obs-logging-tenant/logging-curator5-5.8
+    source: registry.redhat.io/openshift-logging/logging-curator5-rhel9
+EOF
+
+    idms_head="apiVersion: config.openshift.io/v1
+kind: ImageDigestMirrorSet
+metadata:
+  name: logging-qe
+spec:
+  imageDigestMirrors:"
+
+    icsp_head="apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: logging-qe
+spec:
+  repositoryDigestMirrors:"
+
+    mirror_set_clo=""
+    mirror_set_lo=""
+    mirror_set_eo=""
+
+    if [[ $REG_BREW_ENABLED == "true" ]]; then
+	    mirror_set_clo=/tmp/mirror_set_brew_clo.yaml
+	    mirror_set_lo=/tmp/mirror_set_brew_lo.yaml
+	    mirror_set_eo=/tmp/mirror_set_brew_eo.yaml
+    fi
+
+    if [[ "$REG_STAGE_ENABLED" == "true" ]];then
+	    mirror_set_clo=/tmp/mirror_set_stage_clo.yaml
+	    mirror_set_lo=/tmp/mirror_set_stage_lo.yaml
+	    mirror_set_eo=/tmp/mirror_set_stage_eo.yaml
+    fi
+
+    if [[ "$KONFLUX_ENABLED" == "true" ]];then
+	    mirror_set_clo=/tmp/mirror_set_konflux_clo.yaml
+	    mirror_set_lo=/tmp/mirror_set_konflux_lo.yaml
+	    mirror_set_eo=/tmp/mirror_set_konflux_eo.yaml
+    fi
+
+    # Needn't create IDMS/ICSP, return 0
+    if [[ $mirror_set_clo == "" && $mirror_set_clo == "" && $mirror_set_clo == "" ]]; then
+        echo "skip: no ICSP/IDMS "
+        return 0
+    fi
+
+    kube_minor_version=$(oc version --output=json |jq -r '.serverVersion.minor')
+    # use IDMS for cluster > 4.13(kubeverion=1.26)
+    if [[ "$kube_minor_version" -gt "26" ]];then
+        echo "${idms_head}" >/tmp/logging-mirror-set.yaml
+    else
+        echo "${icsp_head}" >/tmp/logging-mirror-set.yaml
+    fi
+
+
+    if [[ $LOGGING_INDEX_IMAGE == "" ]]; then
+        if [[ $CLO_INDEX_IMAGE != "" ]];then
+            cat  $mirror_set_clo>>/tmp/logging-mirror-set.yaml
+        fi
+        if [[ $LO_INDEX_IMAGE != "" ]];then
+            cat $mirror_set_lo>>/tmp/logging-mirror-set.yaml
+        fi
+        if [[ $EO_INDEX_IMAGE != "" ]];then
+            cat $mirror_set_eo>>/tmp/logging-mirror-set.yaml
+        fi
     else
-        echo "!!! fail to create the ICSP"
-        return 1
+        cat  $mirror_set_clo>>/tmp/logging-mirror-set.yaml
+        cat  $mirror_set_lo>>/tmp/logging-mirror-set.yaml
+        cat  $mirror_set_eo>>/tmp/logging-mirror-set.yaml
+    fi
+
+    ## Replace RHEL9 to RHEL10 in feature here.
+    if [[ $LOG_VERSION != "" ]];then
+	#replace 6.3 to 6-3
+	konflux_image_suffix=${LOG_VERSION//./-}
+	sed -i "s/v6-3$/v${konflux_image_suffix}/" /tmp/logging-mirror-set.yaml
+    fi
+    ## we may need to replace RHEL9 to RHEL10 in feature here.
+    # if [[ $$LOG_VERSION >> xxx ]]; then
+    #	sed -i "s/RHEL9/RHEL10" /tmp/logging-mirror-set.yaml
+    # fi
+    #
+
+    run_command "oc apply --overwrite=true -f /tmp/logging-mirror-set.yaml"
+    if [[ $? -eq "0" ]];then
+       CLUSTER_UPDATED=true
     fi
```

Comment:
The step should fail if the command fails.

Reply:
Fixed